api/src/shared/infrastructure/repositories/challenge-repository.js

```diff
@@ -136,8 +137,7 @@ export async function findActiveFlashCompatible({
   const hasCoreScope =
     !complementaryCertificationKey || complementaryCertificationKey === ComplementaryCertificationKeys.CLEA;
 
-  const cacheKey = `findActiveFlashCompatible({ scope: ${hasCoreScope ? Frameworks.CORE : complementaryCertificationKey}, locale: ${locale}, accessibilityAdjustmentNeeded: ${accessibilityAdjustmentNeeded}})`;
-
+  const cacheKey = `findActiveFlashCompatible({ scope: ${hasCoreScope ? Frameworks.CORE : complementaryCertificationKey}, locale: ${locale}, accessibilityAdjustmentNeeded: ${accessibilityAdjustmentNeeded} })`;
```

Comment:
Voir pour te synchro avec @Jeyffrey mais je propose de passer la cacheKey en mode versionId ? Pour éviter des collisions de référentiels en mémoire et aussi supprimer cette "logique métier" dans la couche infra (je trouve que ce n'est pas sa place)

```suggestion
  const cacheKey = `findActiveFlashCompatible({ versionId: ${versionId}, locale: ${locale}, accessibilityAdjustmentNeeded: ${accessibilityAdjustmentNeeded} })`;
```

Reply:
Cette modif est peut etre pas a faire ici dans cette PR mais dans une autre, ou alors temporairement avoir les deux (complementaryKey + versionId) le temps de la bascule